### PR TITLE
feat: refactor changelog platforms

### PR DESCRIPTION
**Pull request by setchy**
## Changes

Refactor changelog source logic in preparation for new source types being added.

## Context

Split from #22094

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by secustor — APPROVED** [review]

**Review by viceice — COMMENTED** [review]
did you test on GitHub and gitlab? so everything works as now after merge?

File: lib/workers/repository/update/pr/changelog/source-github.ts [hunk 1]
```diff
@@ -1,213 +1,80 @@
...
+        logger.warn(
+          { manager, packageName, sourceUrl },
+          'No github.com token has been configured. Skipping release notes retrieval'
         );
+        return { isValid: false, error: 'MissingGithubToken' };
       }
-      changelogReleases.unshift(release);
+      logger.debug(
+        { manager, packageName, sourceUrl },
+        'Repository URL does not match any known github hosts - skipping changelog retrieval'
```
Comment (MindTooth) [inline lib/workers/repository/update/pr/changelog/source-github.ts:74]:
```suggestion
        'Repository URL does not match any known GitHub hosts - skipping changelog retrieval'
```

**Review by MindTooth — COMMENTED** [review]
Minor nits.

**Comment by setchy** [comment]
> did you test on GitHub and gitlab? so everything works as now after merge?

That's correct.  Test repository with GitHub and GitLab packages - https://github.com/setchy/renovate-changelog-refactor

**Comment by setchy** [comment]
> Minor nits.

@MindTooth - let's do that in a separate PR so we correct all nit `GitHub` permutations

**Review by viceice — APPROVED** [review]

**Comment by viceice** [comment]
converted to feature to have a single release, so we can more easily revert if anything goes wrong

**Pull request merged**